File: java/create_campaign.java
import okhttp3.*;

import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        OkHttpClient client = new OkHttpClient().newBuilder().build();

        MediaType mediaType = MediaType.parse("application/json");
        RequestBody body = RequestBody.create(mediaType, "{\n    \"label\": \"My Campaign publipostage\",\n    \"sender\": \"AROLITEC\",\n    \"contacts\": [\n        {\"age\": \"12\", \"numero\": \"2250747189171\", \"name\": \"Martial\"},\n        {\"age\": \"24\", \"numero\": \"2250708651918\", \"name\": \"Rita\"},\n        {\"age\": \"32\", \"numero\": \"225076827512\", \"name\": \"Arolitec\"}\n    ],\n    \"content\": \"Hello {{name}}, votre numéro est {{numero}}\"\n}");

        Request request = new Request.Builder()
                .url("<REPLACE_WITH_PROD_URL>/v1/campaigns")
                .method("POST", body)
                .addHeader("Authorization", "Bearer <YOUR_TOKEN>")
                .addHeader("Content-Type", "application/json")
                .build();

        Response response = client.newCall(request).execute();
        String responseBody = response.body().string();
        System.out.println(responseBody);
    }
}

File: csharp/create_campaign.cs
string prodUrl = "<REPLACE_WITH_PROD_URL>";
string token = "<YOUR_TOKEN>";

var client = new RestClient($"{prodUrl}/v1/campaigns");
client.Timeout = -1;

var request = new RestRequest(Method.POST);
request.AddHeader("Authorization", $"Bearer {token}");
request.AddHeader("Content-Type", "application/json");

var requestBody = new {
    label = "My Campaign publipostage",
    sender = "SMS INFO",
    contacts = new[] {
        new { age = "12", numero = "2250000000000", name = "Martial" },
        new { age = "24", numero = "2250000000000", name = "Rita" },
        new { age = "32", numero = "2260000000000", name = "Arolitec" }
    },
    content = "Hello {{name}}, votre numéro est {{numero}}"
};

request.AddJsonBody(requestBody);

IRestResponse response = client.Execute(request);
Console.WriteLine(response.Content);
